### PR TITLE
Change open portal hyperlinks

### LINKED ISSUE #7170: Change open portal hyperlinks
### Overview of feature request

The `rad app graph` functionality in the CLI can output hyperlinks for Azure resources. It does this by printing the hyperlink next to the resource. 

```txt
Displaying application: demo

Name: demo (Applications.Core/containers)
Connections:
  demo -> db (Applications.Datastores/redisCaches)
Resources:
  demo (apps/Deployment)
  demo (core/Secret)
  demo (core/Service)
  demo (core/ServiceAccount)
  demo (rbac.authorization.k8s.io/Role)
  demo (rbac.authorization.k8s.io/RoleBinding)

Name: db (Applications.Datastores/redisCaches)
Connections:
  demo (Applications.Core/containers) -> db
Resources:
  redis-k5bzgvxs7rwqa (Microsoft.Cache/redis) open portal
```

The suggestion is to make the resource name by the hyperlink instead of a separate `open portal` text.

### Acceptance criteria

```txt
Displaying application: demo

Name: demo (Applications.Core/containers)
Connections:
  demo -> db (Applications.Datastores/redisCaches)
Resources:
  demo (apps/Deployment)
  demo (core/Secret)
  demo (core/Service)
  demo (core/ServiceAccount)
  demo (rbac.authorization.k8s.io/Role)
  demo (rbac.authorization.k8s.io/RoleBinding)

Name: db (Applications.Datastores/redisCaches)
Connections:
  demo (Applications.Core/containers) -> db
Resources:
  redis-k5bzgvxs7rwqa (Microsoft.Cache/redis)
```

In this example `redis-k5bzgvxs7rwqa` would be a hyperlink.

### Additional context

`rad app graph` currently only supports hyperlinks for Azure. Supporting hyperlinks for AWS or other resources is tracked separately.

### Would you like to support us?

- [ ] Yes, I would like to support you

[AB#11160](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/11160)

### PR DESCRIPTION
**Pull request by gpltaylor**
# Description
Allow Connection Graph output resources name to be a link within the console output.

## Type of change

- This pull request fixes a bug in Radius and has an approved issue (issue link required).
- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

Fixes: #7170

### CODE REVIEW

File: pkg/cli/cmd/app/connections/display_test.go
Comment (gpltaylor) [inline pkg/cli/cmd/app/connections/display_test.go:132]: I was thinking about using "html/template" and a tml file. This "expected" String is a little ugly but maybe simpler?

Comment (rynowak) [reply pkg/cli/cmd/app/connections/display_test.go:132]: You could also use a plain text file (eg: `testdata/expected-display-output.txt`). Do we need templating? It looks like the expected output is fully static.

**Comment by rynowak** [comment]
Hey @gpltaylor thanks for helping with this! It looks like the PR is on the right track. I left a suggestion about testing. I'll leave it up to you whether you want to refactor or not. Let me know when you're done and we'll re-review the PR.

**Comment by rynowak** [comment]
In the mean time I've approved the functional tests.

File: pkg/cli/cmd/app/connections/display_test.go
Comment (gpltaylor) [reply pkg/cli/cmd/app/connections/display_test.go:132]: I dislike the string concatenation, I don't want a file IO and templates are pointless (here).... hmm

* Current - concatenation and mixing string operations " ` is ugly
* Templates - As this is static there is no point
* File - The overhead of IO ops is excessive - unpresidented within the code base

I will have a think today...

**Comment by gpltaylor** [comment]
@rynowak @radius-project/radius-core-team
I'm happy with the work done. Can you please review this PR and merge if you're happy to do so.

File: pkg/cli/cmd/app/connections/display_test.go
Comment (rynowak) [reply pkg/cli/cmd/app/connections/display_test.go:132]: By the way we use files in tests for expected output all the time. File IO definitely isn't a concern for me in tests.

**Comment by rynowak** [comment]
Can you rebase this? There are merge conflicts with `main`. If you help let me know.

**Review by rynowak — DISMISSED** [review]
Code changes look good. Thanks @gpltaylor. This will to have conflicts resolved before it goes in.

**Review by kachawla — DISMISSED** [review]

**Comment by gpltaylor** [comment]
@rynowak I think I managed to Rebase and get my code up-to-date. 
Please let me know if this is ok.

**Review by rynowak — APPROVED** [review]
Looks great! Thanks for helping with this.

**Comment by rynowak** [comment]
I'll merge this after the functional tests pass.

**Pull request merged**
